docs/docs/life-and-times-of-a-gatsby-build.md

```diff
@@ -2,7 +2,305 @@
+success createPages - 0.016 s
+success createPagesStatefully - 0.079 s
+success onPreExtractQueries - 0.025 s
+success update schema - 0.041 s
+success extract queries from components - 0.333 s
+success write out requires - 0.020 s
+success write out redirect data - 0.019 s
+success Build manifest and related icons - 0.141 s
+success onPostBootstrap - 0.130 s
+⠀
+info bootstrap finished - 3.674 s
+⠀
+success run static queries - 0.057 s — 3/3 89.08 queries/second
+success run page queries - 0.033 s — 5/5 347.81 queries/second
+- success start webpack server - 1.707 s — 1/1 6.06 pages/second
++ success Generating image thumbnails — 6/6 - 1.059 s
++ success Building production JavaScript and CSS bundles - 8.050 s
++ success Rewriting compilation hashes - 0.021 s
++ success Building static HTML for pages - 0.852 s — 4/4 23.89 pages/second
++ info Done building in 16.143999152 sec
+```
+
+_**Note**: the output of `gatsby develop` and `gatsby build` can vary based on plugins you've installed that can tap into the lifecycle and the [Gatsby reporter](/docs/node-api-helpers/#reporter). The above output is from running the default Gatsby starter._
+
+There is only one difference in the bootstrap phase where HTML and CSS is deleted to prevent problems with previous builds. In the build phase, the build command skips setting up a dev server and goes into compiling the assets.
+
+By omitting these later steps, `gatsby develop` can speed up your ability to make edits with features like [hot module replacement](/docs/glossary#hot-module-replacement). It also saves time with the more CPU intensive processes that aren't necessary to perform while in develoment.
+
+A [cache](/docs/glossary#cache) is also used that is invalidated on every change to a `gatsby-*.js` file (like `gatsby-node.js`, or `gatsby-config.js`) or a dependency.
+
+## What happens when you run `gatsby build`?
+
+To see the code where many of these processes are happening, refer to the code and comments in the [`build`](https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby/src/commands/build.js) and [`bootstrap`](https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby/src/bootstrap/index.js) files of the repository.
+
+A Node process is what is powering things behind the scenes when you run the `gatsby build` command. The process of converting assets and pages into HTML that can be rendered in a browser by a [server-side](docs/glossary#server-side) language like Node.js is referred to as server-side rendering. Since Gatsby is building everything ahead of time, this creates your entire site with all of the data your pages need all at once. Then when the site is deployed, it doesn't need to be running on a server because everything has been gathered up and combined by Gatsby.
+
+**Note**: because Gatsby apps still run React in the browser, you can still fetch data from other sources at [runtime](/docs/glossary#runtime) like you would in a normal React app
```

Comment:
```suggestion
**Note**: because Gatsby apps still run React in the browser, you can still fetch data from other sources at [runtime](/docs/glossary#runtime) like you would in a normal React app.
```

Reply:
is resolved. i close this conversation